src/main/java/com/example/WatchNext/controller/UserController.java

```diff
@@ -0,0 +1,64 @@
+package com.example.WatchNext.controller;
+
+import com.example.WatchNext.model.Role;
+import com.example.WatchNext.model.Users;
+import com.example.WatchNext.payload.request.SignupRequest;
+import com.example.WatchNext.payload.response.MessageResponse;
+import com.example.WatchNext.repositories.RoleRepository;
+import com.example.WatchNext.repositories.UserRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.web.bind.annotation.*;
+
+@CrossOrigin(origins = "*", maxAge = 3600)
+@RestController
+@RequestMapping("/api/auth")
+public class UserController {
+    private UserRepository userRepository;
+    private RoleRepository roleRepository;
+    private PasswordEncoder encoder;
+
+    @Autowired
+    public UserController(UserRepository userRepository,
+                          RoleRepository roleRepository, PasswordEncoder encoder) {
+        this.userRepository = userRepository;
+        this.roleRepository = roleRepository;
+        this.encoder = encoder;
+    }
+
+    @PostMapping("/signup")
```

Comment:
Why do you implement the signup here?
You have all the dependencies to AuthController...also you have there the signin and reset-password, why do you choose to split in this way?